torch/_inductor/compile_fx.py

```diff
@@ -1258,6 +1259,11 @@ def codegen_and_compile(
                     inputs_to_check=inputs_to_check,
                 )
                 metrics_helper = metrics.CachedMetricsHelper()
+
+                # We are going to start code generating runtime asserts, so make sure
+                # you don't start adding new ones in the lowering process
+                graph.freeze_runtime_asserts()
+
```

Comment:
freezing should happen righe before we start codegen of forward and not after 
codegen start with graph.run(*example_inputs)